code/game/objects/effects/particle_holder.dm

```diff
@@ -59,3 +59,46 @@
 /// See [https://www.byond.com/docs/ref/#/{notes}/particles] for position documentation
 /obj/effect/abstract/particle_holder/proc/set_particle_position(list/pos)
 	particles.position = pos
+
+///Stores an index of particles and their instances for reusing : list(particle_type = list(INSTANCES BABEEYYYY))
+GLOBAL_LIST_EMPTY(shared_particle_holders)
+
+///Give a particle effect to an atom (or turf the definition is a lie) from a global list. Use this if you have a lot of particle holders on someones
+///screen and client performance is a problem. This way it's only calculated x amount of times and you can replicate it without putting too much extra burden on clients
+/proc/get_shared_particle_effect(atom/atom, pool_size, particle_effect, particle_flags = NONE)
+	if(!(particle_effect in GLOB.shared_particle_holders)) //no particles yet so make them
+		var/list/particles = list()
+		for(var/i in 1 to pool_size)
+			particles += new /obj/effect/abstract/particle_holder_shared(null, particle_effect, particle_flags)
+
+		GLOB.shared_particle_holders[particle_effect] = particles
+
+	var/obj/effect/abstract/particle_holder_shared/holder = pick(GLOB.shared_particle_holders[particle_effect])
+	holder.apply_particles_to(atom)
+
+///Many particle slow? Have one or two particle and copy paste everywhere. Similair to particle_holder but meant for having multiple vis_locs
+/obj/effect/abstract/particle_holder_shared
+	anchored = TRUE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	layer = ABOVE_ALL_MOB_LAYER
+	vis_flags = VIS_INHERIT_PLANE
+
+	/// Holds info about how this particle emitter works
+	/// See \code\__DEFINES\particles.dm
+	var/particle_flags = NONE
+
+/obj/effect/abstract/particle_holder_shared/Initialize(mapload, particle_path = /particles/smoke, particle_flags = NONE)
+	. = ..()
+
+	src.particle_flags = particle_flags
+	particles = new particle_path
+
+///Apply the particle effect to x obj. Turfs are also fine ignore the typecasting HAHAHHAHAHAHAH i hate byond
+/obj/effect/abstract/particle_holder_shared/proc/apply_particles_to(atom/movable/add_to)
+	add_to.vis_contents += src
+	RegisterSignal(add_to, COMSIG_PARENT_QDELETING, PROC_REF(on_del))
+
+/obj/effect/abstract/particle_holder_shared/proc/on_del(atom/movable/remove_from)
+	SIGNAL_HANDLER
+
+	remove_from.vis_contents -= src
```

Comment:
The signal stuff is redundant here. vis_contents is manually cleared on Destroy(). We handle it for ya, don't need to worry about managing that relationship